### ISSUE TITLE
not keyword in media queries not handles correct and causing incorrect media query removal

### ISSUE THREAD

**Issue by ryuran** [pocketjoso/penthouse#195]
With this parameter `keepLargerMediaQueries: false` on small screen this mq `@media not screen and (min-width: 48em)` is matching but properties in it are not kept by penthouse.

When I enable the property `keepLargerMediaQueries: true`, properties in it are kept in critical CSS.

Whereas It's not a « Larger Media Query ».

I think it’s because it’s negative.

**Comment by pocketjoso** [comment]
The relevant code [is here](https://github.com/pocketjoso/penthouse/blob/master/src/non-matching-media-query-remover.js#L55-L59). The default settings are to only keep the `screen` value for media type (or if the media type is omitted). Indeed there's a bug with the new `keepLargerMediaQueries` option that omits this check, so that non-matching media types are still kept.

Fixing this bug would mean that your `@media not screen` would _always_ be removed. Would that be good for your use case?

(if so, feel free to open a PR for it, or I'll get to it eventually)

**Comment by ryuran** [comment]
```css
@media not screen and (min-width: 48em)
```
in css it means
```
@media not (screen and (min-width: 48em))
```
Because not always refer to the entire mq. (cf: https://developer.mozilla.org/en-US/docs/Web/CSS/Media_Queries/Using_media_queries#not)

So I need content of this MQ on my critical css mobile.
Currently I can’t if I keep `keepLargerMediaQueries: true`.

I will look at it.

**Comment by pocketjoso** [comment]
I had a look at the [media query parsing library](https://github.com/ericf/css-mediaquery), it looks correct.

I believe the only "bug" in the code here is `keepLargerMediaQueries` takes precedent over the other checks. I will open a PR for that now.

**Comment by pocketjoso** [comment]
Seeing another bug with the `not` keyword, working on a fix.

**Comment by pocketjoso** [comment]
So the PR I just opened (https://github.com/pocketjoso/penthouse/pull/200) fixes the bugs I found @ryuran. However note that the mq you listed will as I said now _always_ be removed, as the  media type value is assumed to be `screen` and hence this will never match:
```css
@media not (screen and (min-width: 48em))
```

**Comment by pocketjoso** [comment]
So just to clarify, unless I'm mistaken, what you said originally is _false_:
> small screen this mq @media not screen and (min-width: 48em) is matching

**Comment by ryuran** [comment]
matching by browser, but not matching by penthouse

**Comment by pocketjoso** [comment]
> @media not (screen and (min-width: 48em))

Should not be matched by any desktop browser. What browser and what OS are you on?

**Comment by ryuran** [comment]
This pen can be clearer : https://codepen.io/ryuran/pen/XNPVVP

On small viewport, `#test1` is green and penthouse should keep this matching media query if width parameter is smaller than 50em (320px by exemple)

**Comment by pocketjoso** [comment]
Okay I see your problem now - the test case helps. Will have to look into this going forward.

**Comment by pocketjoso** [comment]
Should be fixed now via #300 thanks to @raxbg ! 👏 

Available via penthouse@2.2.4.
